src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java

```diff
@@ -48,12 +48,44 @@ public interface SonarLintExtendedLanguageServer extends LanguageServer {
   CompletableFuture<SonarLintExtendedLanguageClient.ConnectionCheckResult> checkConnection(ConnectionCheckParams connectionId);
 
   class ConnectionCheckParams {
+    @Nullable
     private String connectionId;
 
+    @Nullable
+    private String token;
+
+    @Nullable
+    private String organization;
```

Comment:
Sometimes we merge these two fields and call it `serverUrlOrOrganisationKey`

Reply:
Firstly I had it implemented that way but didn't come up with a reliable way of detecting whether the `serverUrlOrOrganisationKey` actually belongs to SQ or SC (which I need to determine to construct the `ValidateConnectionParams`) - Do we already do that somewhere in language server?

Reply:
There is `isSonarCloudAlias(String serverUrl)` in `ServerConnectionSettings.java` but I'm not sure it is any help for your case